database/import_test.go

```diff
@@ -0,0 +1,324 @@
+					{"536870912", "17e0aefc0154e0a3cdc4a837c66d9c0e0f0e4a44a703fd6e654e8cbc62c0b28f", "4081063765", "402796026", "1542307497"},
+				},
+				previousBlockHash:  "000000000000000001f34f5eb45827af756e757498039f43ff6f7585c97f4d16",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "255327261802219463033558368",
+				rowIndex:           556761,
+				numberOfBlocks:     10,
+			},
+			expectedBlock: &dto.DbBlockHeader{
+				Height:        556770,
+				Hash:          "00000000000000000005569f09a80c66c8ebf514fdd1c03e803799c2420a4f5a",
+				Version:       536870912,
+				MerkleRoot:    "17e0aefc0154e0a3cdc4a837c66d9c0e0f0e4a44a703fd6e654e8cbc62c0b28f",
+				Timestamp:     timestampInLocalTime("2018-11-15 19:44:57+01:00"),
+				Bits:          402796026,
+				Nonce:         4081063765,
+				State:         "LONGEST_CHAIN",
+				Chainwork:     "2166624730970898396303",
+				CumulatedWork: "255349410425588691745638430",
+				PreviousBlock: "0000000000000000005013e7cc2889ada8b01f24dfc325d1398be82197fc623b",
+			},
+			expectedErrorMessage: "",
+		},
+		{
+			testID: 1,
+			name:   "newer block, should return valid block header",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock: &dto.DbBlockHeader{
+				Height:        833233,
+				Hash:          "00000000000000000676a9b9cdb44820a04c780ca152737124e36341b6c4cdd2",
+				Version:       536870912,
+				MerkleRoot:    "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885",
+				Timestamp:     timestampInLocalTime("2024-02-26 14:33:43+01:00"),
+				Bits:          403300437,
+				Nonce:         3035389718,
+				State:         "LONGEST_CHAIN",
+				Chainwork:     "478151526252246136711",
+				CumulatedWork: "409554917150373038158892043",
+				PreviousBlock: "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+			},
+			expectedErrorMessage: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		var result = []dto.DbBlockHeader{}
+		var err error
+		for i := 0; i < tc.data.numberOfBlocks; i++ {
+			block, err := PrepareRecord(tc.data.blockRecord[i], tc.data.previousBlockHash, tc.data.blockHasher, tc.data.cumulatedChainWork, tc.data.rowIndex)
+			if err != nil {
+				t.Errorf("Error while preparing record: %v", err)
+			}
+			result = append(result, *block)
+			tc.data.previousBlockHash = block.Hash
+			tc.data.cumulatedChainWork = block.CumulatedWork
+			tc.data.rowIndex++
+		}
+		assert.Equal[dto.DbBlockHeader](t, result[tc.data.numberOfBlocks-1], *tc.expectedBlock)
+		assert.NoError(t, err)
+	}
```

Comment:
This way, it will display result for each test case separately, so if one would fail, we will know which one.
```suggestion
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			var result []dto.DbBlockHeader
			var err error
			for i := 0; i < tc.data.numberOfBlocks; i++ {
				block, err := PrepareRecord(tc.data.blockRecord[i], tc.data.previousBlockHash, tc.data.blockHasher, tc.data.cumulatedChainWork, tc.data.rowIndex)
				if err != nil {
					t.Errorf("Error while preparing record: %v", err)
				}
				result = append(result, *block)
				tc.data.previousBlockHash = block.Hash
				tc.data.cumulatedChainWork = block.CumulatedWork
				tc.data.rowIndex++
			}
			assert.Equal[dto.DbBlockHeader](t, result[tc.data.numberOfBlocks-1], *tc.expectedBlock)
			assert.NoError(t, err)
		})
	}
```